syntax = "proto3";

import "google/protobuf/empty.proto"; 
package main;


message Empty {}

message ClientIdentifier {
  int32 client_identifier = 1;
}

message MessageReceived {
  string response = 1;
}

message MessageFormat {
  string first_name = 1;
  int32 client_identifier = 2;
  string message_text = 3;
  int32 server_port_number = 4;
}

message ClientName {
  string first_name = 1;
}

message PortNumber {
  int32 port_number = 1;
}


service ChatService {
  rpc SendMessage(MessageFormat) returns (MessageReceived);
  rpc GetMessage(google.protobuf.Empty) returns (stream MessageFormat);
  rpc GetPortNumber(google.protobuf.Empty) returns (PortNumber);
  rpc GetClientIdentifier(ClientName) returns (ClientIdentifier);
}
